dora/core/common/src/main/java/alluxio/uri/UfsUrl.java

```diff
@@ -0,0 +1,285 @@
+    Preconditions.checkArgument(preprocessingPathList.size() <= 2,
+        "There are multiple schemes, the input may contain more than one path, "
+            + "current version only support inputting one path each time.");
+    // Now, there are two condition. size = 1 -> without scheme; size = 2 -> with a scheme.
+    if (preprocessingPathList.size() == 1)  {
+      // If without scheme, set default scheme to local, i.e. "file".
+      String[] rootDirArray = rootDir.split(SCHEME_SEPERATOR);
+      if (rootDirArray.length == 1) {
+        scheme = "alluxio";
+      } else {
+        scheme = rootDirArray[0];
+      }
+      authorityAndPath = preprocessingPathList.get(0);
+    } else {
+      // preprocessingPathList.size() == 2, i.e. the ufsPath has one scheme.
+      scheme = preprocessingPathList.get(0);
+      authorityAndPath = preprocessingPathList.get(1);
+    }
+    Preconditions.checkNotNull(scheme, "scheme is empty, please input again");
+    Preconditions.checkNotNull(authorityAndPath, "authority or path is empty, please input again");
+    Preconditions.checkArgument(scheme.equalsIgnoreCase("file")
+            || scheme.equalsIgnoreCase("s3")
+            || scheme.equalsIgnoreCase("hdfs")
+            || scheme.equalsIgnoreCase("alluxio"),
+        "Now UfsUrl only support alluxio, file(local), s3, and hdfs, please check scheme first");
+    int indexOfFirstSlashAfterAuthority = authorityAndPath.indexOf(PATH_SEPERATOR);
+    if (indexOfFirstSlashAfterAuthority == -1)  {
+      // If index is 0, the authorityString will be empty.
+      indexOfFirstSlashAfterAuthority = 0;
+    }
+    String authorityString = authorityAndPath.substring(0, indexOfFirstSlashAfterAuthority);
+    String pathString = authorityAndPath.substring(indexOfFirstSlashAfterAuthority);
+
+    if (scheme.equalsIgnoreCase("file") || scheme.equalsIgnoreCase("alluxio")
+        || scheme.equalsIgnoreCase("s3") || scheme.equalsIgnoreCase("hdfs")) {
+      pathString = rootDir + pathString;
+    } else {
+      System.out.println("Other scheme are not supported currently.");
+      return;
+    }
+    String[] arrayOfPathString = pathString.split(PATH_SEPERATOR);
+    List<String> pathComponentsList = Arrays.asList(arrayOfPathString);
+    for (String partPath : pathComponentsList) {
+      if (!partPath.isEmpty())  {
+        mDepth++;
+      }
+    }
+    mProto = UfsUrlMessage.newBuilder()
+        .setScheme(scheme)
+        .setAuthority(authorityString)
+        .addAllPathComponents(pathComponentsList)
+        .build();
+  }
+
+  public boolean hasScheme() {
+    return mProto.hasScheme();
+  }
+
+  public String getScheme() {
+    return mProto.getScheme();
+  }
+
+  public boolean hasAuthority() {
+    return mProto.hasAuthority();
+  }
+
+  public Authority getAuthority() {
+    return Authority.fromString(mProto.getAuthority());
+  }
+
+  public String getAuthorityString() {
+    return mProto.getAuthority();
+  }
```

Comment:
the caller should be able to call `Authority.toString` to achieve this. I'd prefer this utility method to be removed.

Reply:
remain line 119 optional<String> like above comment.